crates/holochain/src/sweettest/sweet_conductor.rs

```diff
@@ -542,6 +553,53 @@ impl SweetConductor {
         let connectivity = covering(rng, all.len(), s);
         crate::conductor::p2p_agent_store::exchange_peer_info_sparse(all, connectivity).await;
     }
+
+    /// Wait for at least one gossip round to have completed for the given cell
+    ///
+    /// Note that this is really a crutch. If gossip starts fast enough then this is unnecessary
+    /// but that doesn't necessarily happen. Waiting for gossip to have started before, for example,
+    /// waiting for something else like consistency is useful to ensure that communication has
+    /// actually started.
+    pub async fn require_initial_gossip_activity_for_cell(
+        &self,
+        cell: &SweetCell,
+        timeout: Duration,
```

Comment:
It's kind of misleading to be checking gossip activity for a cell, since gossip applies to the entire conductor. I see why you did it though: because `network_info` now requires an agent key (via #2183). I think this is unfortunate and we shouldn't need to specify an agent key to get gossip info. I'd like us (the team) to consider refactoring this, but just making note of it for now.

Reply:
Agreed, happy to update this if the `network_info` interface changes. Equally happy to work on making that change